deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/main/java/org/deeplearning4j/clustering/lsh/LSH.java

```diff
@@ -0,0 +1,77 @@
+package org.deeplearning4j.clustering.lsh;
+
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.BaseAccumulation;
+
+/**
+ * This interface gathers the minimal elements for an LSH implementation
+ *
+ * See chapter 3 of :
+ * _Mining Massive Datasets_, Anand Rajaraman and Jeffrey Ullman
+ * http://www.mmds.org/
+ *
+ */
+public interface LSH {
+
```

Comment:
Could we make this an op maybe? It looks really similar. It feels like we could fit this in to the Op type.

Reply:
The hashing part maybe. How would you implement the query ?